org.alloytools.alloy.classic.provider/src/main/java/org/alloytools/alloy/classic/provider/Relation.java

```diff
@@ -0,0 +1,202 @@
+package org.alloytools.alloy.classic.provider;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.alloytools.alloy.core.api.IAtom;
+import org.alloytools.alloy.core.api.IRelation;
+import org.alloytools.alloy.core.api.ITuple;
+import org.alloytools.alloy.core.api.Solution;
+
+public class Relation implements IRelation {
+
+    final Solution solution;
```

Comment:
similar to my other question: why does a relation need a back pointer to a solution?